### ISSUE TITLE
DateTime::getLastErrors only returns false when no date functions have been called

### ISSUE THREAD

**Issue by marcinmdev** [php/doc-en#1351]
From manual page: https://php.net/datetime.getlasterrors

---
```
PHP 8.0.2 (cli) (built: Feb  9 2021 19:20:59) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.2, Copyright (c) Zend Technologies
```

```
php > var_dump(DateTime::createFromFormat('d/m/Y','01/01/2022')::getLastErrors());
array(4) {
  ["warning_count"]=>
  int(0)
  ["warnings"]=>
  array(0) {
  }
  ["error_count"]=>
  int(0)
  ["errors"]=>
  array(0) {
  }
}

```

**Comment by cmb69** [comment]
If no date function had been called prior to calling `date_get_last_errors()`, [`false` is returned](https://3v4l.org/rvMCX). It's probably a good idea to document that (and maybe change the behavior sometime in the future).

**Comment by othercorey** [comment]
@kamil-tekiela This should be fixed with the last patch

**Comment by derickr** [comment]
We fixed this behaviour in PHP 8.2, at which time we also documented this in the changelog: https://www.php.net/manual/en/datetimeimmutable.getlasterrors.php#refsect1-datetimeimmutable.getlasterrors-changelog
